cmd/skaffold/app/cmd/runner.go

```diff
@@ -46,32 +46,30 @@ import (
 var createRunner = createNewRunner
 
 func withRunner(ctx context.Context, out io.Writer, action func(runner.Runner, []*latest.SkaffoldConfig) error) error {
-	runner, config, err := createRunner(out, opts)
+	runner, runCtx, config, err := createRunner(out, opts)
 	if err != nil {
 		return err
 	}
 
 	err = action(runner, config)
 
-	return alwaysSucceedWhenCancelled(ctx, err)
+	return alwaysSucceedWhenCancelled(ctx, runCtx, err)
 }
 
 // createNewRunner creates a Runner and returns the SkaffoldConfig associated with it.
-func createNewRunner(out io.Writer, opts config.SkaffoldOptions) (runner.Runner, []*latest.SkaffoldConfig, error) {
+func createNewRunner(out io.Writer, opts config.SkaffoldOptions) (runner.Runner, *runcontext.RunContext, []*latest.SkaffoldConfig, error) {
```

Comment:
Does the runContext need to be explicitly returned, or could the runner.Runner have a `Context()` method?

Reply:
I added a `ErrConfig` to keep scope limited.